sycl/source/detail/device_global_map_entry.cpp

```diff
@@ -26,21 +38,24 @@ DeviceGlobalUSMMem::~DeviceGlobalUSMMem() {
          "MZeroInitEvent has not been cleaned up.");
 }
 
-std::optional<RT::PiEvent>
-DeviceGlobalUSMMem::getZeroInitEvent(const plugin &Plugin) {
+OwnedPiEvent DeviceGlobalUSMMem::getZeroInitEvent(const plugin &Plugin) {
   std::lock_guard<std::mutex> Lock(MZeroInitEventMutex);
   // If there is a zero-init event we can remove it if it is done.
-  if (MZeroInitEvent.has_value() &&
-      get_event_info<info::event::command_execution_status>(
-          *MZeroInitEvent, Plugin) == info::event_command_status::complete) {
-    Plugin.call<PiApiKind::piEventRelease>(*MZeroInitEvent);
-    MZeroInitEvent = {};
+  if (MZeroInitEvent.has_value()) {
+    if (get_event_info<info::event::command_execution_status>(
+            *MZeroInitEvent, Plugin) == info::event_command_status::complete) {
+      Plugin.call<PiApiKind::piEventRelease>(*MZeroInitEvent);
+      MZeroInitEvent = {};
+      return OwnedPiEvent(Plugin);
+    } else {
+      return std::move(OwnedPiEvent(*MZeroInitEvent, Plugin));
```

Comment:
This code doesn't compile with clang.
https://github.com/intel/llvm/actions/runs/4244006981/jobs/7377519363

Reply:
https://github.com/intel/llvm/pull/8437